packages/excubiae/contracts/extensions/SemaphoreExcubia.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import {Excubia} from "../Excubia.sol";
+import {ISemaphore} from "@semaphore-protocol/contracts/interfaces/ISemaphore.sol";
+
+/// @title Semaphore Excubia Contract
+/// @notice This contract extends the Excubia contract to integrate with the Semaphore protocol.
+/// It verifies the passerby Semaphore group membership proofs to grant access through the gate.
+/// @dev To allow only specific Semaphore identities from a group, the contract stores the specific group identifier.
+/// To avoid identities from passing twice, nullifiers are stored upon successful verification of the proofs.
+contract SemaphoreExcubia is Excubia {
+    /// @notice The Semaphore contract interface.
+    ISemaphore public immutable SEMAPHORE;
+    /// @notice The specific group identifier that proofs must match to pass the gate.
+    /// @dev Used as a `scope` to ensure consistency during proof membership verification.
+    uint256 public immutable GROUP_ID;
+
+    /// @notice Mapping to track which nullifiers have been used to avoid passing the
+    /// gate twice using the same Semaphore identity.
+    /// @dev The nullifier is derived from the hash of the secret and group identifier,
+    /// ensuring that the same identity cannot be registered twice for the same group.
+    mapping(uint256 => bool) public passedNullifiers;
+
+    /// @notice Error thrown when the group identifier does not match the expected one.
+    error InvalidGroup();
+
+    /// @notice Error thrown when the proof is invalid.
+    error InvalidProof();
+
+    /// @notice Error thrown when the proof scope does not match the expected group identifier.
+    error UnexpectedScope();
+
+    /// @notice Constructor to initialize with target Semaphore contract and specific group identifier.
+    /// @param _semaphore The address of the Semaphore contract.
+    /// @param _groupId The group identifier that proofs must match.
+    constructor(address _semaphore, uint256 _groupId) {
+        if (_semaphore == address(0)) revert ZeroAddress();
+
+        SEMAPHORE = ISemaphore(_semaphore);
+
+        if (ISemaphore(_semaphore).groupCounter() < _groupId) revert InvalidGroup();
```

Comment:
```suggestion
        if (ISemaphore(_semaphore).groupCounter() <= _groupId) revert InvalidGroup();
```

I think this should be `<=`. If `groupCounter` is 1 (meaning there's 1 group only with id 0), and `_groupId` is 1, this check will pass. Can you double-check?

Reply:
this condition is correct - based on your example:

groupCounter = 1 (we have one valid group with groupId equal to 0)

- 1 < 0 no revert
- 1 < 1 revert (correct, since only groupId = 0 is valid - there's just one group)

suppose to create another group, groupCounter = 2

- 2 < 0 no revert
- 2 < 1 no revert
- 2 < 2 revert ...

Reply:
1 < 1 and 2 < 2 are still false, like the other conditions, so it shouldn't revert 🤔

Reply:
ooo you're right!